### PR TITLE
[Fix #5757] Add option to allow parentheses around multiline conditions

### LINKED ISSUE #5757: Configuration for disabling `ParenthesesAroundCondition` in multi-line conditions
Hello!

I don't like extra parentheses, like many of us.

But sometimes they seem to me appropriate. Especially, in cases with multi-line conditions.

--------

## Expected behavior

I don't want to get offense from `ParenthesesAroundCondition` for multi-line conditions.

## Actual behavior

I'm getting.

## Steps to reproduce the problem

```ruby
if (
  one.update(one_values) &&
  two.update(two_values)
)
  return true
end
```

I can't write so. I need to align:

```ruby
if one.update(one_values) &&
   two.update(two_values)
  return true
end
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.1 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by Darhazer**
-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@
 * [#5286](https://github.com/bbatsov/rubocop/issues/5286): Fix `Lint/SafeNavigationChain` not detecting chained operators after block. ([@Darhazer][])
 * Fix bug where `Lint/SafeNavigationConsistency` registers multiple offenses for the same method call. ([@rrosenblum][])
 
+### Changes
+* [#5757](https://github.com/bbatsov/rubocop/issues/5757): Add `AllowInMultilineConditions` option to `Style/ParenthesesAroundCondition` cop. ([@Darhazer][])
```
Comment (bbatsov) [inline CHANGELOG.md]: You forgot the blank line after the section heading.

File: lib/rubocop/cop/style/parentheses_around_condition.rb [hunk 2]
```diff
@@ -71,6 +70,16 @@ def message(node)
...
         end
+
+        def parens_allowed?(cond)
+          parens_required?(cond) ||
+            (safe_assignment?(cond) && safe_assignment_allowed?) ||
+            (cond.multiline? && allow_multiline_conditions?)
+        end
+
+        def allow_multiline_conditions?
+          cop_config['AllowInMultilineConditions']
```
Comment (bbatsov) [inline lib/rubocop/cop/style/parentheses_around_condition.rb]: You should also mention the new config option in the cop examples/description.

File: lib/rubocop/cop/style/parentheses_around_condition.rb [hunk 3]
```diff
@@ -71,6 +70,16 @@ def message(node)
           article = kw == 'while' ? 'a' : 'an'
           "Don't use parentheses around the condition of #{article} `#{kw}`."
         end
+
+        def parens_allowed?(cond)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/parentheses_around_condition.rb]: I wonder where `cond`, `node` or `cond_node` is the best param name here, but that's not a big deal.

**Comment by Darhazer** [comment]
Updated

**Comment by bbatsov** [comment]
👍

**Pull request merged**